### PR TITLE
[Fixes #7739] An empty map cannot be saved

### LINKED ISSUE #7739: An empty map cannot be saved
@allyoucanmap commented on [Wed May 26 2021](https://github.com/GeoNode/geonode-mapstore-client/issues/205)

It seems that a new map must have at least one layer from geonode to be saved correctly without the layer the thumbnail process seems to fail.
This is the error returned by the save request:

```
detail: "Retry in 180s: ThumbnailError(\"Thumbnail generation couldn't determine a BBOX for: AA by admin.\")"
```

steps to reproduce:
- create a new map without add layer
- open the burger menu and open the Save As modal
- add a title and click to SaveAs...

_current result_:

- the save request fail with this error
```
detail: "Retry in 180s: ThumbnailError(\"Thumbnail generation couldn't determine a BBOX for: AA by admin.\")"
```

_expected result_:

- there should not be error caused by the thumbnail process

### PR DESCRIPTION
**Pull request by marthamareal**
<Include a few sentences describing the overall goals for this Pull Request>
References #7739

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Review by giohappy — CHANGES_REQUESTED** [review]
fix conflicts

**Comment by marthamareal** [comment]
done

File: geonode/thumbs/thumbnails.py [hunk 1]
```diff
@@ -107,6 +107,18 @@ def create_thumbnail(
...
 
     compute_bbox_from_datasets = False
+    is_map_with_datasets = True
+
+    if isinstance(instance, Map):
+        viable_datasets = []
+        for map_dataset in instance.datasets:
+            if map_dataset.visibility and map_dataset.local and map_dataset.ows_url:
+                # Skip no visible layer and incorrectly defined remote layer encountered (no OWS URL defined)
+                viable_datasets.append(map_dataset)
```
Comment (afabiani) [inline geonode/thumbs/thumbnails.py:117]: So, if `viable_datasets` won't be used for other purposes, it does not make really sense to do the whole loop, store the layers into a list just to set a `boolean` later on.

Let's please optimize this code.

**Review by afabiani — CHANGES_REQUESTED** [review]

File: geonode/thumbs/thumbnails.py (hunk 1 above)
Comment (marthamareal) [reply geonode/thumbs/thumbnails.py]: makes sense. updated

**Review by afabiani — APPROVED** [review]

**Pull request merged**